pkg/manifests/manifests.go

```diff
@@ -2087,6 +2087,28 @@ func (f *Factory) MetricsServerDeployment(kubeletCABundle *v1.ConfigMap, tlsSecr
 		return dep, nil
 	}
 
+	r := newErrMapReader(requestheader)
```

Comment:
from the definition, errMapReader would expose the first occurred error in errMapReader.err. Does this mean that if there are several errors we will just miss others than the first one? just a nit, but wanted to double check.

Reply:
I am not sure why doc string is put that way, we use same for adapter as well. 
IIUC it means if first error occurs itself it will stop execution and won't check for more errors?